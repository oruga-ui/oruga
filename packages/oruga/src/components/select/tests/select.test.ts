import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OSelect from "@/components/select/Select.vue";

describe("OSelect tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OSelect);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("select");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
