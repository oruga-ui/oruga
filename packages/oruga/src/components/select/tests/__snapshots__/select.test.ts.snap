// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OSelect tests > render correctly 1`] = `
"<div class="o-ctrl-sel" data-oruga="select">
  <!--v-if--><select id="mocked" data-oruga-input="select" class="o-sel o-sel-arrow" autocomplete="off">
    <!--v-if-->
    <!--
                @slot Override the options, default is options prop
            -->
  </select>
  <!--v-if-->
</div>"
`;
